### ISSUE TITLE
Begin thinking about reserved names

### ISSUE THREAD

**Issue by themeblvd** [themeblvd/backroad#8]
In moving ahead with #6, and we start growing other functionality and potentially clashing endpoints we should start thinking about implementing reserved name lists.

For example, if #6 adds `/first`, `/last` and `/count` endpoints to querying users, we could potentially have clash because I've chosen to allow users to be queried by the username like:

`/api/v1/users/<username>/`

So if the following query brings back the count of users:

`/api/v1/users/count/`

Then, a user created named `count` would cause some issues.

To sum up, we need a reserved names list that will apply to usernames and article slugs when being created. To start:

* count
* first
* last
